cpp/open3d/data/Downloader.cpp

```diff
@@ -102,135 +105,76 @@ static size_t WriteDataCb(void* ptr, size_t size, size_t nmemb, FILE* stream) {
     return written;
 }
 
-static std::string GetAbsoluteFilePath(const std::string& url,
-                                       const std::string& output_file_path,
-                                       const std::string& output_file_name) {
-    std::string file_name;
-    if (output_file_name.empty()) {
-        file_name = utility::filesystem::GetFileNameWithoutDirectory(url);
-    } else {
-        file_name = output_file_name;
+void DownloadFromURL(const std::string& url,
+                     const std::string& sha256,
+                     const std::string& prefix,
+                     const std::string& data_root) {
+    // Always print URL to inform the user. If the download fails, the user
+    // knows the URL.
+    utility::LogInfo("Downloading {}", url);
+
+    // Sanity checks.
+    if (sha256.size() != SHA256_DIGEST_LENGTH * 2) {
+        utility::LogError("Invalid sha256 length {}, expected to be {}.",
+                          sha256.size(), SHA256_DIGEST_LENGTH * 2);
     }
-
-    std::string file_path;
-    if (output_file_path.empty()) {
-        file_path = LocateDataRoot();
-    } else {
-        file_path = output_file_path;
+    if (prefix.empty()) {
+        utility::LogError("Download prefix cannot be empty.");
     }
 
-    // It will create the directory hierarchy if not present.
-    if (!utility::filesystem::DirectoryExists(file_path)) {
-        utility::filesystem::MakeDirectoryHierarchy(file_path);
+    // Resolve path.
+    const std::string resolved_data_root =
+            data_root.empty() ? LocateDataRoot() : data_root;
+    const std::string file_dir = resolved_data_root + "/" + prefix;
```

Comment:
[TODO] we may add a `utility::filesystem::join_path(...)` function, since it is used in many places, and the `prefix` may contain extra `/`. I will try to add that.

Reply:
Yes. `std::filesystem` already have that. We can directly use it later.